src/index.js

```diff
@@ -297,14 +297,15 @@ Multiaddr.prototype.equals = function equals (addr) {
  * // {family: 'IPv4', address: '127.0.0.1', port: '4001'}
  */
 Multiaddr.prototype.nodeAddress = function nodeAddress () {
-  if (!this.isThinWaistAddress()) {
-    throw new Error('Multiaddr must be "thin waist" address for nodeAddress.')
-  }
-
   const codes = this.protoCodes()
   const parts = this.toString().split('/').slice(1)
+
+  if (parts.length < 4) {
+    throw new Error('Multiaddr must be a valid address for nodeAddress.')
+  }
+
   return {
-    family: (codes[0] === 41) ? 'IPv6' : 'IPv4',
+    family: (codes[0] === 41 || codes[0] === 55) ? 'IPv6' : 'IPv4',
```

Comment:
This seems still incorrect (and was before your change as well) as https://nodejs.org/api/http.html#http_http_request_options_callback says it has to be a `number`, not a string.